aml-development/frontend
# fix(pki): various bug fixes for login via PKI

**Description**
Allow users to access the site via both PKI (prod) and username/password (dev)

**Acceptance Criteria**
1. Users can login via PKI.  Site behaves correctly for that user.
2. Users can login/logout via username/password.  Site behaves correctly for that user.

**How to Test**
You only need to pull this branch.

_PKI (Prod)_
This is all done in Firefox using the backend + nginx configuration on `berlin` (backend SME's machine).

1. Install certs.  Follow: https://github.com/aml-development/dev-tools/tree/master/certs#firefox-steps (use port `8085` instead of `4443` and `http` instead of `https` to access certs).
2. Run frontend using the following command `npm run dev -- --env.API_URL="https://berlin:4443" --env.NODE_ENV="production"`
3. Close any existing browser windows, then open a new tab and go to `http://localhost:8080/`.
4. When prompted for a cert, use the desired user you want to login as.
5. Ensure that the site loads with the correct user (correct listings on the mainpage, correct profile, etc).  Ensure that you do not have a logout button in the hamburger menu and you do not go to a login page.
6. Repeat 3-5 for different users.

_Username/Password (Dev)_
You can test this in any browser.

1. Run frontend using the following command `npm run dev`.
2. Close any existing browser windows, then open a new tab and go to `http://localhost:8080/`.
3. Ensure that you are directed to a login page.  Enter the username and password of your choice and click Login.
4. Ensure that the site loads with the correct user (correct listings on the mainpage, correct profile, etc).  
5. Ensure that you have a logout button in the hamburger menu and clicking it logs you out successfully.
6. Repeat 2-5 for different users.
